### ISSUE TITLE
Configuring HyperParameters

### ISSUE THREAD

**Issue by s-bernard** [oracle/tribuo#344]
Hi, I want to train a logistic regression with Tribuo.

I successfully used LogisticRegressionTrainer() but I cannot find where to configure the hyper parameters and in particular the regularization parameter. I want to be sure it adds a bias and it used a L2 regularization.

Also, do you intend to add a LBFGS optimizer?

Thanks.

**Comment by Craigacp** [comment]
`LogisticRegressionTrainer` is a simplified subclass for `LinearSGDTrainer` and all the hyperparameters are exposed there. However Tribuo always adds a bias, and we don't implement L2 regularization through the model yet, though some of the optimizers may implicitly implement regularization. There's a rewrite coming for those classes in v5 which will add more features like better stopping criteria and regularization via weight decay, but v5 isn't going to happen before the end of the year.

**Comment by s-bernard** [comment]
OK, thanks. Will you release any preview before?

**Comment by Craigacp** [comment]
No, but the head of `main` should always work so you'll be able to build it yourself.

**Comment by s-bernard** [comment]
True. Thanks.
